src/Caliburn.Micro.Platform/Platforms/Maui/Windows/MauiPlatformProvider.cs

```diff
@@ -0,0 +1,242 @@
+        /// The WindowManager marks that element as a framework-created element so that it can determine what it created vs. what was intended by the developer.
+        /// Calling GetFirstNonGeneratedView allows the framework to discover what the original element was.
+        /// </remarks>
+        public virtual object GetFirstNonGeneratedView(object view) 
+        {
+            return View.GetFirstNonGeneratedView(view);
+        }
+
+        private static readonly DependencyProperty PreviouslyAttachedProperty = DependencyProperty.RegisterAttached(
+            "PreviouslyAttached",
+            typeof (bool),
+            typeof (MauiPlatformProvider),
+            null
+            );
+
+        /// <summary>
+        /// Executes the handler the fist time the view is loaded.
+        /// </summary>
+        /// <param name="view">The view.</param>
+        /// <param name="handler">The handler.</param>
+        public virtual void ExecuteOnFirstLoad(object view, Action<object> handler) 
+        {
+            //var element = view as FrameworkElement;
+            //if (element != null && !(bool) element.GetValue(PreviouslyAttachedProperty)) {
+            //    element.SetValue(PreviouslyAttachedProperty, true);
+            //    View.ExecuteOnLoad(element, (s, e) => handler(s));
+            //}
+        }
+
+        /// <summary>
+        /// Executes the handler the next time the view's LayoutUpdated event fires.
+        /// </summary>
+        /// <param name="view">The view.</param>
+        /// <param name="handler">The handler.</param>
+        public virtual void ExecuteOnLayoutUpdated(object view, Action<object> handler) 
+        {
+            //var element = view as FrameworkElement;
+            //if (element != null) {
+            //    View.ExecuteOnLayoutUpdated(element, (s, e) => handler(s));
+            //}
+        }
+
+        /// <summary>
+        /// Get the close action for the specified view model.
+        /// </summary>
+        /// <param name="viewModel">The view model to close.</param>
+        /// <param name="views">The associated views.</param>
+        /// <param name="dialogResult">The dialog result.</param>
+        /// <returns>
+        /// An <see cref="Action" /> to close the view model.
+        /// </returns>
+        /// <exception cref="System.NotImplementedException"></exception>
+        public virtual Func<CancellationToken, Task> GetViewCloseAction(object viewModel, ICollection<object> views, bool? dialogResult)
+        {
+            foreach (var contextualView in views) {
+                var viewType = contextualView.GetType();
+#if MAUI
+                var closeMethod = viewType.GetRuntimeMethod("Close", new Type[0]);
+#else
+                var closeMethod = viewType.GetMethod("Close", new Type[0]);
+#endif
+                if (closeMethod != null)
+                    return ct => {
+#if !MAUI
```

Comment:
Do you need the conditional compliation if the class is an maui folder?

Reply:
No I think this is left over from the UWP provider playing double duty. Will clean it up.